wolfcrypt/src/port/Renesas/renesas_tsip_util.c

```diff
@@ -0,0 +1,654 @@
+
+    ret = R_TSIP_Sha1HmacGenerateInit(&_handle, &key_index);
+    
+    if (ret == TSIP_SUCCESS)
+        ret = R_TSIP_Sha1HmacGenerateUpdate(&_handle, (uint8_t*)myInner, 
+                                                        (uint32_t)innerSz);
+    
+    if (ret == TSIP_SUCCESS)
+        ret = R_TSIP_Sha1HmacGenerateUpdate(&_handle, (uint8_t*)in, sz);
+    
+    if (ret == TSIP_SUCCESS)
+        ret = R_TSIP_Sha1HmacGenerateFinal(&_handle, digest);
+    
+    /* unlock hw */
+    tsip_hw_unlock();
+    
+    return ret;
+}
+
+/* Sha256Hmac */
+int tsip_Sha256Hmac(const struct WOLFSSL *ssl, const byte *myInner, 
+            word32 innerSz, const byte *in, word32 sz, byte *digest, 
+            word32 verify)
+{
+    tsip_hmac_sha_handle_t _handle;
+    int ret;
+    tsip_hmac_sha_key_index_t key_index;
+
+    if ( (ssl->options.side == WOLFSSL_CLIENT_END && !verify) ||
+            (ssl->options.side == WOLFSSL_SERVER_END &&  verify) )
+        XMEMCPY(key_index.value, ssl->keys.tsip_client_write_MAC_secret,
+                                                    sizeof(key_index.value));
+       else
+        XMEMCPY(key_index.value, ssl->keys.tsip_server_write_MAC_secret,
+                                                    sizeof(key_index.value));
+
+    if ((ret = tsip_hw_lock()) != 0) {
+        WOLFSSL_MSG("hw lock failed\n");
+        return ret;
+    }
+    
+    ret = R_TSIP_Sha256HmacGenerateInit(&_handle, &key_index);
+    
+    if (ret == TSIP_SUCCESS)
+        ret = R_TSIP_Sha256HmacGenerateUpdate(&_handle, (uint8_t*)myInner, 
+                                                                    innerSz);
+    
+    if (ret == TSIP_SUCCESS)
+        ret = R_TSIP_Sha256HmacGenerateUpdate(&_handle, (uint8_t*)in, sz);
+    
+    if (ret == TSIP_SUCCESS)
+        ret = R_TSIP_Sha256HmacGenerateFinal(&_handle, digest);
+    
+    /* unlock hw */
+    tsip_hw_unlock();
+    
+    return ret;
+}
+
+/* generate Verify Data based on master secret */
+int tsip_generateVerifyData(const byte *ms, /* master secret */
+                            const byte *side, const byte *handshake_hash,
+                            byte *hashes /* out */)
+{
+    int ret ;
+    uint32_t l_side = R_TSIP_TLS_GENERATE_CLIENT_VERIFY;
+    
+    if (XSTRNCMP((const char*)side, (const char*)tls_server, FINISHED_LABEL_SZ)
+                                                                           == 0)
+    {
+        l_side = R_TSIP_TLS_GENERATE_SERVER_VERIFY;
+    }
+    
+    if ((ret = tsip_hw_lock()) == 0) {
+        ret = R_TSIP_TlsGenerateVerifyData(l_side, (uint32_t*)ms,
+                       (uint8_t*)handshake_hash, hashes/* out */);
+        if (ret != TSIP_SUCCESS) {
+            WOLFSSL_MSG("R_TSIP_TlsGenerateSessionKey failed\n");
+        }
+    }
+    /* unlock hw */
+    tsip_hw_unlock();
+    
+    return ret;
+}
+
+/* generate keys for TLS communication */
+int tsip_generateSeesionKey(struct WOLFSSL *ssl)
```

Comment:
Do we need to sanitize input arguments against NULL here?

Reply:
Yes, we do. Add sanity-check the function and some functions too.